library_website/views.py

```diff
@@ -0,0 +1,47 @@
+from django.http import Http404
+from django.shortcuts import render
+from wagtail.images.models import Image
+from wagtail.models import Site
+
+
+def library_404_view(request, exception=Http404):
+
+    site_name = Site.find_for_request(request).site_name
+
+    class Link:
+
+        def __init__(self, text, url):
+            self.text = text
+            self.url = url
+
+    def context_404(site_name):
+        if site_name == "Public":
+            return {
+                "404_page_template_switcher":
+                "base/public_base.html",
+                "links": [
+                    Link("Ask a Librarian", "/ask-librarian/"),
+                    Link("Library Digital Collections", "/collex/"),
+                    Link("Library Homepage", "/")
+                ],
+                "helptext1":
+                "Let us know that we have a missing page.",
+                "helptext2":
+                " We will do our best to rectify the issue."
+            }
+        else:
+            return {
+                "404_page_template_switcher":
+                "intranethome/intranet_home_page.html",
+                "koala": Image.objects.get(title="Adventure Koala"),
```

Comment:
This should probably have a safe fallback for when the image isn't present and then in the template it should only call the image tag if the image is present.